Purple Dot Pre-Order button/scripts.js

```diff
@@ -1,102 +1,140 @@
-let selectedVariant = Tapcart.variables.product.selectedVariant.id;
-let currency = Tapcart.variables.cart.currency;
-const onlyAllowCurrencies = [];
-
-const apiKey = "<YOUR_API_KEY>";
-
-const instockButton = document.querySelector('#in-stock-button');
-const preorderButton = document.querySelector('#preorder-button');
-const dispatchDateDiv = document.querySelector('#waitlist-dispatch-date');
-const waitlistInfoDiv = document.querySelector('#waitlist-info');
-const learnMoreLink = document.querySelector('#learn-more-link');
-
-learnMoreLink.addEventListener('click', () => {
-  Tapcart.actions.openScreen({
-    destination: {
-      type: "web",
-      url: `https://www.purpledotprice.com/embedded-checkout/pre-order-value-prop?hideBackToProduct=true&isAddToPreorder=false&apiKey=${apiKey}&noModal=true&salesChannel=tapcart`
-    }
+wrapWithErrorTracking(() => {
```

Comment:
I think this doesn't handle async errors or errors thrown from event handler, it will only catch errors in the initial sync code inside the function.

I don't know how rigorous this needs to be? Maybe adding an unhandled promise rejection handler would be enough to cover most cases.